Error "unknown type name 'BN_ULONG'" while building project on simulator

When i try to build project in XCode (Version 10.2.1) on simulator i get the error:

> In file included from [path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/pkcs7.h:62:
In file included from [path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/asn1.h:74:
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:314:5: error: unknown type name 'BN_ULONG'
    BN_ULONG *d;                /* Pointer to an array of 'BN_BITS2' bit
    ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:330:5: error: unknown type name 'BN_ULONG'
    BN_ULONG n0[2];             /* least significant word(s) of Ni; (type
    ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:487:22: error: a parameter list without types is only allowed in a function definition
int BN_num_bits_word(BN_ULONG);
                     ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:535:1: error: unknown type name 'BN_ULONG'
BN_ULONG BN_mod_word(const BIGNUM *a, BN_ULONG w);
^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:535:39: error: unknown type name 'BN_ULONG'
BN_ULONG BN_mod_word(const BIGNUM *a, BN_ULONG w);
                                      ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:536:1: error: unknown type name 'BN_ULONG'
BN_ULONG BN_div_word(BIGNUM *a, BN_ULONG w);
^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:536:33: error: unknown type name 'BN_ULONG'
BN_ULONG BN_div_word(BIGNUM *a, BN_ULONG w);
                                ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:537:28: error: unknown type name 'BN_ULONG'
int BN_mul_word(BIGNUM *a, BN_ULONG w);
                           ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:538:28: error: unknown type name 'BN_ULONG'
int BN_add_word(BIGNUM *a, BN_ULONG w);
                           ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:539:28: error: unknown type name 'BN_ULONG'
int BN_sub_word(BIGNUM *a, BN_ULONG w);
                           ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:540:28: error: unknown type name 'BN_ULONG'
int BN_set_word(BIGNUM *a, BN_ULONG w);
                           ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:541:1: error: unknown type name 'BN_ULONG'
BN_ULONG BN_get_word(const BIGNUM *a);
^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:557:37: error: unknown type name 'BN_ULONG'
int BN_mod_exp_mont_word(BIGNUM *r, BN_ULONG a, const BIGNUM *p,
                                    ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:596:24: error: unknown type name 'BN_ULONG'
void BN_consttime_swap(BN_ULONG swap, BIGNUM *a, BIGNUM *b, int nwords);
                       ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:914:1: error: unknown type name 'BN_ULONG'
BN_ULONG bn_mul_add_words(BN_ULONG *rp, const BN_ULONG *ap, int num,
^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:914:27: error: unknown type name 'BN_ULONG'
BN_ULONG bn_mul_add_words(BN_ULONG *rp, const BN_ULONG *ap, int num,
                          ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:914:47: error: unknown type name 'BN_ULONG'
BN_ULONG bn_mul_add_words(BN_ULONG *rp, const BN_ULONG *ap, int num,
                                              ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:915:27: error: unknown type name 'BN_ULONG'
                          BN_ULONG w);
                          ^
[path to project]Pods/Headers/Public/OpenSSL-Universal/openssl/bn.h:916:1: error: unknown type name 'BN_ULONG'
BN_ULONG bn_mul_words(BN_ULONG *rp, const BN_ULONG *ap, int num, BN_ULONG w);
^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.

The project is compile successfully on a real device.
I suspect that needs to make changes to "Build Settings". How I can fix this error?

--- a/.gitignore
+++ b/.gitignore
@@ -21,3 +21,6 @@ xcuserdata/
 **/xcshareddata/WorkspaceSettings.xcsettings
 
 Carthage/
+
+*.tar.gz
+*.tar.gz.sha256

--- a/OpenSSL.xcodeproj/project.pbxproj
+++ b/OpenSSL.xcodeproj/project.pbxproj
@@ -14,6 +14,7 @@
 		757359FF22BEBE5800BC52E8 /* opensslconf-x86_64.h in Headers */ = {isa = PBXBuildFile; fileRef = 757359FE22BEBE5800BC52E8 /* opensslconf-x86_64.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		75F1C98D22BE87FB00F44E23 /* shim.h in Headers */ = {isa = PBXBuildFile; fileRef = 75F1C98B22BE87E600F44E23 /* shim.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		75F1C98F22BE881300F44E23 /* shim.h in Headers */ = {isa = PBXBuildFile; fileRef = 75F1C98E22BE881200F44E23 /* shim.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		84BBFB582331B8D300DD11C1 /* opensslconf-i386.h in Headers */ = {isa = PBXBuildFile; fileRef = 84BBFB572331B8D200DD11C1 /* opensslconf-i386.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		9F4A26B9223BD8DB005CB63A /* mdc2.h in Headers */ = {isa = PBXBuildFile; fileRef = 9F4A266E223BD8D5005CB63A /* mdc2.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		9F4A26BA223BD8DB005CB63A /* aes.h in Headers */ = {isa = PBXBuildFile; fileRef = 9F4A266F223BD8D5005CB63A /* aes.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		9F4A26BB223BD8DB005CB63A /* blowfish.h in Headers */ = {isa = PBXBuildFile; fileRef = 9F4A2670223BD8D5005CB63A /* blowfish.h */; settings = {ATTRIBUTES = (Public, ); }; };
@@ -180,6 +181,7 @@
 		757359FE22BEBE5800BC52E8 /* opensslconf-x86_64.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = "opensslconf-x86_64.h"; path = "macos/include/openssl/opensslconf-x86_64.h"; sourceTree = "<group>"; };
 		75F1C98B22BE87E600F44E23 /* shim.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = shim.h; path = macos/include/openssl/shim.h; sourceTree = "<group>"; };
 		75F1C98E22BE881200F44E23 /* shim.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = shim.h; path = ios/include/openssl/shim.h; sourceTree = "<group>"; };
+		84BBFB572331B8D200DD11C1 /* opensslconf-i386.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = "opensslconf-i386.h"; path = "ios/include/openssl/opensslconf-i386.h"; sourceTree = "<group>"; };
 		9F4A265C223BD82B005CB63A /* OpenSSL.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = OpenSSL.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		9F4A266E223BD8D5005CB63A /* mdc2.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = mdc2.h; path = ios/include/openssl/mdc2.h; sourceTree = "<group>"; };
 		9F4A266F223BD8D5005CB63A /* aes.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = aes.h; path = ios/include/openssl/aes.h; sourceTree = "<group>"; };
@@ -443,6 +445,7 @@
 				757359F622BEBE2B00BC52E8 /* opensslconf-arm64.h */,
 				757359F922BEBE2C00BC52E8 /* opensslconf-armv7.h */,
 				757359F722BEBE2B00BC52E8 /* opensslconf-armv7s.h */,
+				84BBFB572331B8D200DD11C1 /* opensslconf-i386.h */,
 				757359F822BEBE2C00BC52E8 /* opensslconf-x86_64.h */,
 				9F4A269C223BD8D8005CB63A /* opensslv.h */,
 				9F4A2683223BD8D6005CB63A /* ossl_typ.h */,
@@ -680,6 +683,7 @@
 				9F4A26E0223BD8DB005CB63A /* pqueue.h in Headers */,
 				9F4A26D6223BD8DB005CB63A /* des_old.h in Headers */,
 				9F4A270E223BD9EB005CB63A /* OpenSSL.h in Headers */,
+				84BBFB582331B8D300DD11C1 /* opensslconf-i386.h in Headers */,
 				9F4A26D1223BD8DB005CB63A /* ts.h in Headers */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;

--- a/build.sh
+++ b/build.sh
@@ -44,12 +44,16 @@ configure() {
 
    if [ "$ARCH" == "x86_64" ]; then
        ${SRC_DIR}/Configure darwin64-x86_64-cc --prefix="${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}" &> "${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}.log"
-       sed -ie "s!^CFLAG=!CFLAG=-isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK} -arch $ARCH -mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} !" "${SRC_DIR}/Makefile"
-       sed -ie "s!^CFLAGS=!CFLAGS=-isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK} -arch $ARCH -mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} !" "${SRC_DIR}/Makefile"
+       sed -ie "s!^CFLAG=!CFLAG=-isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK} -arch $ARCH -mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} -fembed-bitcode !" "${SRC_DIR}/Makefile"
+       sed -ie "s!^CFLAGS=!CFLAGS=-isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK} -arch $ARCH -mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} -fembed-bitcode !" "${SRC_DIR}/Makefile"
+   elif [ "$ARCH" == "i386" ]; then
+       ${SRC_DIR}/Configure darwin-i386-cc --prefix="${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}" &> "${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}-conf.log"
+       sed -ie "s!^CFLAG=!CFLAG=-isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK} -arch $ARCH -mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} -fembed-bitcode !" "${SRC_DIR}/Makefile"
+       sed -ie "s!^CFLAGS=!CFLAGS=-isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK} -arch $ARCH -mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} -fembed-bitcode !" "${SRC_DIR}/Makefile"      
    else
        ${SRC_DIR}/Configure iphoneos-cross -no-asm --prefix="${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}" &> "${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}.log"
-       sed -ie "s!^CFLAG=!CFLAG=-mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} !" "${SRC_DIR}/Makefile"
-       sed -ie "s!^CFLAGS=!CFLAGS=-mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} !" "${SRC_DIR}/Makefile"
+       sed -ie "s!^CFLAG=!CFLAG=-mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} -fembed-bitcode -arch ${ARCH} !" "${SRC_DIR}/Makefile"
+       sed -ie "s!^CFLAGS=!CFLAGS=-mios-simulator-version-min=${DEPLOYMENT_VERSION} -miphoneos-version-min=${DEPLOYMENT_VERSION} -fembed-bitcode -arch ${ARCH} !" "${SRC_DIR}/Makefile"
        perl -i -pe 's|static volatile sig_atomic_t intr_signal|static volatile int intr_signal|' ${SRC_DIR}/crypto/ui/ui_openssl.c
    fi
 }
@@ -69,7 +73,8 @@ build()
    echo "Building for ${SDK##*/} ${ARCH}"
 
    export BUILD_TOOLS="${DEVELOPER}"
-   export CC="${BUILD_TOOLS}/usr/bin/gcc -fembed-bitcode -arch ${ARCH}"
+   export CC="${BUILD_TOOLS}/usr/bin/gcc"
+   export CFLAGS="-fembed-bitcode -arch ${ARCH}"
 
    # Change dir
    cd "${SRC_DIR}"
@@ -80,7 +85,7 @@ build()
 
    if [ "$TYPE" == "ios" ]; then
       # IOS
-      if [ "$ARCH" == "x86_64" ]; then
+      if [ "$ARCH" == "x86_64" ] || [ "$ARCH" == "i386" ]; then
          configure "iPhoneSimulator" $ARCH ${IPHONESIMULATOR_PLATFORM} ${IPHONEOS_SDK_VERSION} ${IPHONEOS_DEPLOYMENT_VERSION} ${BUILD_DIR} ${SRC_DIR}
       else
          configure "iPhoneOS" $ARCH ${IPHONEOS_PLATFORM} ${IPHONEOS_SDK_VERSION} ${IPHONEOS_DEPLOYMENT_VERSION} ${BUILD_DIR} ${SRC_DIR}
@@ -89,8 +94,8 @@ build()
       #OSX
       if [ "$ARCH" == "x86_64" ]; then
          ${SRC_DIR}/Configure darwin64-x86_64-cc --prefix="${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}" &> "${BUILD_DIR}/${OPENSSL_VERSION}-${ARCH}.log"
-         sed -ie "s!^CFLAG=!CFLAG=-isysroot ${SDK} -arch $ARCH -mmacosx-version-min=${OSX_DEPLOYMENT_VERSION} !" "${SRC_DIR}/Makefile"
-         sed -ie "s!^CFLAGS=!CFLAGS=-isysroot ${SDK} -arch $ARCH -mmacosx-version-min=${OSX_DEPLOYMENT_VERSION} !" "${SRC_DIR}/Makefile"
+         sed -ie "s!^CFLAG=!CFLAG=-isysroot ${SDK} -arch $ARCH -mmacosx-version-min=${OSX_DEPLOYMENT_VERSION} -fembed-bitcode !" "${SRC_DIR}/Makefile"
+         sed -ie "s!^CFLAGS=!CFLAGS=-isysroot ${SDK} -arch $ARCH -mmacosx-version-min=${OSX_DEPLOYMENT_VERSION} -fembed-bitcode !" "${SRC_DIR}/Makefile"
       fi
    fi
 
@@ -123,6 +128,10 @@ generate_opensslconfh() {
 # include <openssl/opensslconf-x86_64.h>
 #endif
 
+#if defined(__APPLE__) && defined (__i386__)
+# include <openssl/opensslconf-i386.h>
+#endif
+
 #if defined(__APPLE__) && defined (__arm__) && defined (__ARM_ARCH_7A__)
 # include <openssl/opensslconf-armv7.h>
 #endif
@@ -144,6 +153,7 @@ build_ios() {
    rm -rf ${SCRIPT_DIR}/{ios/include,ios/lib}
    mkdir -p ${SCRIPT_DIR}/{ios/include,ios/lib}
 
+   build "i386" ${IPHONESIMULATOR_SDK} ${TMP_DIR} "ios"
    build "x86_64" ${IPHONESIMULATOR_SDK} ${TMP_DIR} "ios"
    build "armv7"  ${IPHONEOS_SDK} ${TMP_DIR} "ios"
    build "armv7s" ${IPHONEOS_SDK} ${TMP_DIR} "ios"
@@ -154,6 +164,7 @@ build_ios() {
    cp -f ${SCRIPT_DIR}/shim/shim.h ${SCRIPT_DIR}/ios/include/openssl/shim.h
 
    cp -f ${TMP_DIR}/${OPENSSL_VERSION}-x86_64/include/openssl/opensslconf-x86_64.h ${SCRIPT_DIR}/ios/include/openssl
+   cp -f ${TMP_DIR}/${OPENSSL_VERSION}-i386/include/openssl/opensslconf-i386.h ${SCRIPT_DIR}/ios/include/openssl
    cp -f ${TMP_DIR}/${OPENSSL_VERSION}-armv7/include/openssl/opensslconf-armv7.h ${SCRIPT_DIR}/ios/include/openssl
    cp -f ${TMP_DIR}/${OPENSSL_VERSION}-armv7s/include/openssl/opensslconf-armv7s.h ${SCRIPT_DIR}/ios/include/openssl
    cp -f ${TMP_DIR}/${OPENSSL_VERSION}-arm64/include/openssl/opensslconf-arm64.h ${SCRIPT_DIR}/ios/include/openssl

